ocaml/xapi/xapi_psr_util.ml

```diff
@@ -0,0 +1,40 @@
+(* avoiding circular deps *)
+module D = Debug.Make (struct let name = "xapi_psr" end)
+
+module Unixext = Stdext.Unixext
+
+let checkpoint_path = "/var/lib/xcp/psr_cp"
+
+let checkpoint_tmp_path = "/var/lib/xcp/psr_cp.tmp"
+
+let old_pool_secret_backup_path = "/var/lib/xcp/ptoken.old"
+
+let new_pool_secret_backup_path = "/var/lib/xcp/ptoken.new"
+
+let load_psr_pool_secrets () =
+  match
+    ( Sys.file_exists old_pool_secret_backup_path
+    , Sys.file_exists new_pool_secret_backup_path )
+  with
+  | false, false ->
+      ()
+  | false, true | true, false ->
+      D.error "expected both files to exist: %s , %s"
+        old_pool_secret_backup_path new_pool_secret_backup_path ;
+      raise
+        Api_errors.(
+          Server_error
+            (internal_error, ["inconsistent pool secret backup files"]))
+  | true, true ->
+      D.info "loading backup pool secrets from psr" ;
+      let secrets =
+        [old_pool_secret_backup_path; new_pool_secret_backup_path]
+        |> List.map (fun path ->
+               Unixext.string_of_file path |> SecretString.of_string)
+      in
+      (* be careful to load them in the correct order *)
+      Xapi_globs.pool_secrets :=
+        if SecretString.equal (Xapi_globs.pool_secret ()) (List.hd secrets) then
+          secrets
+        else
+          List.rev secrets
```

Comment:
I would rather add a bit of repetition when processing pool secret paths and keep both secrets in separate variables rather than having to do this strange handling.